quinn-h3/src/connection.rs

```diff
@@ -236,16 +240,27 @@ impl ConnectionInner {
     }
 
     fn poll_send(&mut self, cx: &mut Context) -> Result<(), Error> {
-        if let Some(data) = self.inner.pending_control() {
-            self.send_control.push(data);
-        }
-        match Pin::new(&mut self.send_control).poll(cx) {
-            Poll::Ready(Err(err)) => {
-                self.set_error(ErrorCode::CLOSED_CRITICAL_STREAM, format!("{:?}", err));
-                return Err(err);
+        let types = [
+            PendingStreamType::CONTROL,
+            PendingStreamType::ENCODER,
+            PendingStreamType::DECODER,
+        ];
+        for ty in types.iter() {
```

Comment:
Writing `&ty` here will make the explicit derefs unnecessary.

Reply:
Ha, nice !